src/go/rpk/pkg/cli/cmd/start.go

```diff
@@ -533,16 +588,88 @@ func parseFlags(flags []string) map[string]string {
 	return parsed
 }
 
+func parseSeeds(seeds []string) ([]config.SeedServer, error) {
+	seedServers := []config.SeedServer{}
+	for _, s := range seeds {
+		addressID := strings.Split(s, "+")
+		if len(addressID) != 2 {
+			return seedServers, fmt.Errorf(
+				"Couldn't parse seed '%s': Format doesn't"+
+					" conform to %s."+
+					" Missing ID.",
+				s,
+				seedFormat,
+			)
+		}
+		id, err := strconv.Atoi(strings.Trim(addressID[1], " "))
+		if err != nil {
+			return seedServers, fmt.Errorf(
+				"Couldn't parse seed '%s': ID must be an int.",
+				s,
+			)
+		}
+		addr, err := parseAddress(addressID[0])
+		if err != nil {
+			return seedServers, fmt.Errorf(
+				"Couldn't parse seed '%s': %v",
+				s,
+				err,
+			)
+		}
+		if addr == nil {
+			return seedServers, fmt.Errorf(
+				"Couldn't parse seed '%s': empty address",
+				s,
+			)
+		}
+		seedServers = append(
+			seedServers,
+			config.SeedServer{Host: *addr, Id: id},
+		)
+	}
+	return seedServers, nil
+}
+
+func parseAddress(addr string) (*config.SocketAddress, error) {
+	if addr == "" {
+		return nil, nil
+	}
+	hostPort := strings.Split(addr, ":")
+	if len(hostPort) != 2 {
+		// It's just a hostname with no port. Assume 9092.
+		return &config.SocketAddress{
+			Address: strings.Trim(hostPort[0], " "),
```

Comment:
Empty host isn't checked as it is below. This should be reorganised to do the checks once.

I think a test for: `[]string{" +1234"},` should catch it,

Reply:
Good catch!